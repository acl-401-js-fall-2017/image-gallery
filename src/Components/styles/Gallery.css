.Gallery {
    height: 100%;
}

.imgSlide {
    height: 60%;
    width: 80%;
    margin: 1em auto;
    overflow: hidden;
}

.Gallery img {
    height: 90%;

}

.galleryButtons {
    display: flex;
    justify-content: center;
    height: 10%;
}

.galleryButtons h1 {
    width:50%;
    text-align: center;
    background-color: black;
    border-radius: 0.25em;
    color: white;
    margin: 0.75em 1em;
    line-height: 1.5em;
}