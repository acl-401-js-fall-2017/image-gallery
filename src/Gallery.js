import React, { PureComponent } from 'react';
import styled from 'styled-components';

export default class Thumbnail extends PureComponent {
  constructor() {
    super();
    this.state= { index: 0}
    this.history =[];
  }

  setState(state, ignore) {
    if(!ignore) this.history.push(state);
    super.setState(state);
  }

  handleClick = (value) =>{
    console.log('clicked');
    const newState ={
      ...this.state,
      index: this.state.index + value
    }
    this.setState(newState);
  }
  render() {
    const {gallery, shouldDisplay } =this.props
    return( 
      <StyledDiv shouldDisplay={shouldDisplay}>
        <div>
          {this.props.gallery.map((img, i) => (
            <ImgDiv shouldDisplay ={this.state.index === i}>
              <span> {img.title} </span>
              <img style={{ width:'100%' }} key={img._id} src={img.img} alt='image'/>
              <span> {img.description} </span>
              { i !== gallery.length -1 && <span onClick ={()=> this.handleClick(1)}> next </span>}
              {i !== 0 && <span onClick ={()=> this.handleClick(-1)}> previous</span>}
            </ImgDiv>
          ))}
        </div>
      </StyledDiv>
    )
  };
}

const StyledDiv = styled.div`
display:${props => props.shouldDisplay ? 'flex' : 'none'};
`;

const ImgDiv = styled.div`
display:${props => props.shouldDisplay ? 'flex' : 'none'};
flex-direction: column;
margin: 0 10%;
`;

