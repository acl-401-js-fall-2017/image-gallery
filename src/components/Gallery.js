import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import Bunnies from '../images/bunnies';



export default class Gallery extends PureComponent {

    static propTypes = {
      bunnies: PropTypes.arrayOf(PropTypes.object).isRequired
    }

    constructor(){
      super();
      this.state = {
        bunnies: Bunnies,
        picIndex: 0
      };
    }


    nextImage = ({ target }) => {
      this.setState(prevState => {
        let picIndex;
        if (target.id === 'prevButton') {
          picIndex = prevState.picIndex - 1;
        }
        if (target.id === 'nextButton') {
          picIndex = prevState.picIndex + 1;
        }
        return { picIndex };
      });
    }

    render() {
      const bunny = this.props.bunnies[this.state.picIndex];

      return (
        <div className="gallery">
          <img src={bunny.url} alt={bunny.title} height="100px"/>
          <h4>{bunny.title}</h4>
          <h6>{bunny.description}</h6>

          {this.state.picIndex !== 0 && (
            <button type="button" value="Previous" id="prevButton" onClick={this.nextImage}>
          Previous
            </button>
          )}

          {this.state.picIndex !== this.state.bunnies.length - 1  && (
            <button type="button" value="Next" id="nextButton" onClick={this.nextImage}>
          Next
            </button>
          )}
          {/* Might want to add remove button here */}
        </div>
      );
    }
}