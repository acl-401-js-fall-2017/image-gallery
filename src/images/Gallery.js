import React, { PureComponent } from 'react';
import styled from 'styled-components';
import { loadGallery, deleteImg } from '../utils/actions';
import imgApi from '../utils/services/imgApi';


export default class Gallery extends PureComponent {
  state = { 
    index: 0, 
    gallery:[]
  }

  async componentDidMount() {
    const img = await imgApi.get();
    const newState = loadGallery(this.state, img);
    this.setState(newState);
  }

  handleDelete = async id => {
    await imgApi.remove(id);
    const newState = deleteImg(this.state, id);
    this.setState(newState);
  }


  handleClick = (value) => {
    console.log('clicked');
    const newState = {
      ...this.state,
      index: this.state.index + value
    };
    this.setState(newState);
  }

  render() {
    return( 
      <StyledDiv>
        <div>
          {this.state.gallery.map((img, i) => (
            <ImgDiv key={img._id} shouldDisplay ={this.state.index === i}>
              <span> {img.title} </span>
              <DeleteDiv onClick={() => this.handleDelete(img._id)}>X</DeleteDiv>
              <img style={{ width:'100%' }} src={img.url} alt=''/>
              <span> {img.description} </span>
              { i !== this.state.gallery.length -1 && <span onClick ={()=> this.handleClick(1)}> next </span>}
              {i !== 0 && <span onClick ={()=> this.handleClick(-1)}> previous</span>}
            </ImgDiv>
          ))}
        </div>
      </StyledDiv>
    );
  }
}

const StyledDiv = styled.div`
display: 'flex';
`;

const ImgDiv = styled.div`
display:${props => props.shouldDisplay ? 'flex' : 'none'};
flex-direction: column;
margin: 0 10%;
`;

const DeleteDiv = styled.div`
margin-left: 90%;
text-align: center;
border: 1px solid black;
`;