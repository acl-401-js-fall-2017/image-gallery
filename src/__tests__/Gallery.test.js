import React from 'react';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Bunnies from '../images/bunnies';
import Gallery from '../components/Gallery';

describe('List', () => {
  it('shows bunnies gallery format', () => {
    const wrapper = shallow(<Gallery bunnies={Bunnies}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});