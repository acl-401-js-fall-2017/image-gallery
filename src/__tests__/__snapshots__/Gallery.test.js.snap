// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List shows bunnies gallery format 1`] = `
<div
  className="gallery"
>
  <img
    alt="Best Bunny"
    height="100px"
    src="https://s3.amazonaws.com/f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
  />
  <h4>
    Best Bunny
  </h4>
  <h6>
    Just the cutest color on a bunny ever
  </h6>
  <button
    id="nextButton"
    onClick={[Function]}
    type="button"
    value="Next"
  >
    Next
  </button>
</div>
`;
